atc/db/check_factory.go

```diff
@@ -143,7 +142,7 @@ func (c *checkFactory) TryCreateCheck(logger lager.Logger, checkable Checkable,
 	parentType, found := resourceTypes.Parent(checkable)
 	if found {
 		if parentType.Version() == nil {
-			return nil, false, errors.New("parent type has no version")
+			return nil, false, fmt.Errorf("resource's parent type '%s' has no version", parentType.Name())
```

Comment:
```suggestion
			return nil, false, fmt.Errorf("resource type '%s' has no version", parentType.Name())
```
Would it be more aligned to Concourse docs this way?

Reply:
I think it's important to have the word `parent` in the error message. It makes it clear what the relationship is between the resource page the user is currently on and the _other_ (parent) resource that is causing the error.

In general we have a tendency to write very robotic sounding error messages, probably because we're in the middle of writing code which usually reads very robotic. This PR is trying to make the error message more readable to the human that will eventually read it.

Reply:
In our doc we don't say "resource's parent type" but just "a resource", "a resource type" or "a type for the resource". In our common sence "resource's parent type" = "resource type" while we are concourse dev but for just concourse users the "parent" might cause a confusion.

More specifically IMHO the error message should align with the first sentence of [Resource Type's page](https://concourse-ci.org/resource-types.html#resource-types) in our doc, where it says
> Each resource in a pipeline has a type. 

However I am not good at UX so not 100% sure others would take the message the same way. How about summoning @matthewpereira to get some inputs? Thx!

Reply:
That convinced me!